core/src/main/java/org/apache/carbondata/core/datastore/page/VarLengthColumnPageBase.java

```diff
@@ -176,7 +179,7 @@ private static ColumnPage getDecimalColumnPage(TableSpec.ColumnSpec columnSpec,
     rowOffset.putInt(counter, offset);
 
     VarLengthColumnPageBase page;
-    if (unsafe) {
+    if (unsafe && !meta.isFillCompleteVector()) {
```

Comment:
many place check like this, can we make a function for it and make it more readable by give proper function name?

Reply:
ok

Reply:
ok